### PR TITLE
[ Magiclysm ] Pickaxe List Fix #70710

### LINKED ISSUE #70710: [Magiclysm] Enchanted pickaxes don't count as a pickaxe for contrustion tasks.
### Describe the bug

Pickaxe +1 doesn't count as a pickaxe for construction related tasks.  It does function correctly for auto-mining and manual mining by activating the tool.

### Attach save file

[Try not to cheat-trimmed.tar.gz](https://github.com/CleverRaven/Cataclysm-DDA/files/13850374/Try.not.to.cheat-trimmed.tar.gz)

### Steps to reproduce

Spawn a pickaxe +1
Enter construction menu and select any mining task.
See that your enchanted pickaxe doesn't count as a pickaxe.

### Expected behavior

Enchanted pickaxes should still be pickaxes

### Screenshots

image

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19045.3803 (22H2)
- Game Version: 2a65f10 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    Slowdown Fungal Growth [no_fungal_growth],
    Aftershock [aftershock],
    Bionic Professions [package_bionic_professions],
    Blaze Industries [blazeindustries],
    Magiclysm [magiclysm],
    Mind Over Matter [mindovermatter],
    Niam's Mods [niam_mods],
    Xedra Evolved [xedra_evolved],
    DinoMod [DinoMod],
    Megafauna [megafauna],
    Bombastic Perks [bombastic_perks],
    Extra Mutated Scenarios [extra_mut_scens],
    SpeedyDex [speedydex],
    Stats Through Kills [stats_through_kills],
    Stats Through Skills [StatsThroughSkills],
    Tamable Wildlife [Tamable_Wildlife]
]

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Maleclypse**
#### Summary
None

#### Purpose of change
Fix #70710

#### Describe the solution
Makes a pickaxe list.  Should make a jackhammer standard while I'm here but I didn't.

#### Describe alternatives you've considered
Making a jackhammer list too

### CODE REVIEW

File: data/mods/innawood/recipes/recipe_deconstruction.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
     "activity_level": "MODERATE_EXERCISE",
     "skill_used": "fabrication",
     "time": "5 m",
-    "tools": [ [ [ "pickaxe", -1 ], [ "bronze_pickaxe", -1 ], [ "jackhammer", 1 ], [ "elec_jackhammer", 50 ] ] ],
+    "tools": [ [ [ [ "surface_heat", 1, "LIST" ] ], [ "jackhammer", 1 ], [ "elec_jackhammer", 50 ] ] ],
```
Comment (Karol1223) [inline data/mods/innawood/recipes/recipe_deconstruction.json:16]:
```suggestion
    "tools": [ [ [ [ "pickaxe_list", 1, "LIST" ] ], [ "jackhammer", 1 ], [ "elec_jackhammer", 50 ] ] ],
```
I'm assuming you didn't intend to break rocks with a hotplate?

**Review by Karol1223 — CHANGES_REQUESTED** [review]

File: data/json/requirements/toolsets.json [hunk 2]
```diff
@@ -295,7 +301,7 @@
     "id": "mining_standard",
     "type": "requirement",
     "//": "mining",
-    "tools": [ [ [ "pickaxe", -1 ], [ "bronze_pickaxe", -1 ], [ "jackhammer", 300 ], [ "elec_jackhammer", 3960 ] ] ]
+    "tools": [ [ [ [ "pickaxe_list", 1, "LIST" ] ], [ "jackhammer", 300 ], [ "elec_jackhammer", 3960 ] ] ]
```
Comment (Maleclypse) [inline data/json/requirements/toolsets.json:304]:
```suggestion
    "tools": [ [ [ "pickaxe_list", 1, "LIST" ] ], [ "jackhammer", 300 ], [ "elec_jackhammer", 3960 ] ] ]
```

File: data/json/uncraft/generic.json [hunk 3]
```diff
@@ -118,7 +118,7 @@
     "activity_level": "MODERATE_EXERCISE",
     "skill_used": "fabrication",
     "time": "5 m",
-    "tools": [ [ [ "pickaxe", -1 ], [ "bronze_pickaxe", -1 ], [ "jackhammer", 5 ], [ "elec_jackhammer", 66 ] ] ],
+    "tools": [ [ [ [ "pickaxe_list", 1, "LIST" ] ], [ "jackhammer", 5 ], [ "elec_jackhammer", 66 ] ] ],
```
Comment (Maleclypse) [inline data/json/uncraft/generic.json:121]:
```suggestion
    "tools": [ [ [ "pickaxe_list", 1, "LIST" ], [ "jackhammer", 5 ], [ "elec_jackhammer", 66 ] ] ],
```

File: data/json/requirements/toolsets.json [hunk 4]
```diff
@@ -370,7 +375,7 @@
     "//": "Tools for removing road / sidewalk",
     "qualities": [ { "id": "PRY", "level": 3 } ],
     "tools": [
-      [ [ "pickaxe", -1 ], [ "bronze_pickaxe", -1 ], [ "jackhammer", 140 ], [ "elec_jackhammer", 1848 ] ],
+      [ [ [ "pickaxe_list", 1, "LIST" ] ], [ "jackhammer", 140 ], [ "elec_jackhammer", 1848 ] ],
```
Comment (Maleclypse) [inline data/json/requirements/toolsets.json:378]:
```suggestion
      [ [ "pickaxe_list", 1, "LIST" ], [ "jackhammer", 140 ], [ "elec_jackhammer", 1848 ] ],
```

File: data/json/requirements/toolsets.json [hunk 5]
```diff
@@ -295,7 +301,7 @@
     "id": "mining_standard",
     "type": "requirement",
     "//": "mining",
-    "tools": [ [ [ "pickaxe", -1 ], [ "bronze_pickaxe", -1 ], [ "jackhammer", 300 ], [ "elec_jackhammer", 3960 ] ] ]
+    "tools": [ [ [ "pickaxe_list", 1, "LIST" ], [ "jackhammer", 300 ], [ "elec_jackhammer", 3960 ] ] ]
```
Comment (Maleclypse) [inline data/json/requirements/toolsets.json:304]:
```suggestion
    "tools": [ [ [ "pickaxe_list", -1, "LIST" ], [ "jackhammer", 300 ], [ "elec_jackhammer", 3960 ] ] ]
```

File: data/json/requirements/toolsets.json [hunk 6]
```diff
@@ -370,7 +375,7 @@
     "//": "Tools for removing road / sidewalk",
     "qualities": [ { "id": "PRY", "level": 3 } ],
     "tools": [
-      [ [ "pickaxe", -1 ], [ "bronze_pickaxe", -1 ], [ "jackhammer", 140 ], [ "elec_jackhammer", 1848 ] ],
+      [ [ "pickaxe_list", 1, "LIST" ], [ "jackhammer", 140 ], [ "elec_jackhammer", 1848 ] ],
```
Comment (Maleclypse) [inline data/json/requirements/toolsets.json:378]:
```suggestion
      [ [ "pickaxe_list", -1, "LIST" ], [ "jackhammer", 140 ], [ "elec_jackhammer", 1848 ] ],
```

File: data/json/uncraft/generic.json [hunk 7]
```diff
@@ -118,7 +118,7 @@
     "activity_level": "MODERATE_EXERCISE",
     "skill_used": "fabrication",
     "time": "5 m",
-    "tools": [ [ [ "pickaxe", -1 ], [ "bronze_pickaxe", -1 ], [ "jackhammer", 5 ], [ "elec_jackhammer", 66 ] ] ],
+    "tools": [ [ [ "pickaxe_list", 1, "LIST" ], [ "jackhammer", 5 ], [ "elec_jackhammer", 66 ] ] ],
```
Comment (Maleclypse) [inline data/json/uncraft/generic.json:121]:
```suggestion
    "tools": [ [ [ "pickaxe_list", -1, "LIST" ], [ "jackhammer", 5 ], [ "elec_jackhammer", 66 ] ] ],
```

File: data/mods/innawood/recipes/recipe_deconstruction.json [hunk 8]
```diff
@@ -13,7 +13,7 @@
     "activity_level": "MODERATE_EXERCISE",
     "skill_used": "fabrication",
     "time": "5 m",
-    "tools": [ [ [ "pickaxe", -1 ], [ "bronze_pickaxe", -1 ], [ "jackhammer", 1 ], [ "elec_jackhammer", 50 ] ] ],
+    "tools": [ [ [ [ "pickaxe_list", 1, "LIST" ] ], [ "jackhammer", 1 ], [ "elec_jackhammer", 50 ] ] ],
```
Comment (Maleclypse) [inline data/mods/innawood/recipes/recipe_deconstruction.json:16]:
```suggestion
    "tools": [ [ [ [ "pickaxe_list", -1, "LIST" ] ], [ "jackhammer", 1 ], [ "elec_jackhammer", 50 ] ] ],
```

File: data/json/requirements/toolsets.json [hunk 9]
```diff
@@ -295,7 +301,7 @@
     "id": "mining_standard",
     "type": "requirement",
     "//": "mining",
-    "tools": [ [ [ "pickaxe", -1 ], [ "bronze_pickaxe", -1 ], [ "jackhammer", 300 ], [ "elec_jackhammer", 3960 ] ] ]
+    "tools": [ [ [ "pickaxe_list", -1, "LIST" ], [ "jackhammer", 300 ], [ "elec_jackhammer", 3960 ] ] ]
```
Comment (Maleclypse) [inline data/json/requirements/toolsets.json:304]:
```suggestion
   "tools": [ [ [ "pickaxe", -1 ], [ "bronze_pickaxe", -1 ], [ "jackhammer", 300 ], [ "elec_jackhammer", 3960 ] ] ]
```

File: data/mods/Magiclysm/requirements/crafting.json [hunk 10]
```diff
@@ -3,5 +3,17 @@
...
+  {
+    "id": "pickaxe_list",
+    "type": "requirement",
+    "//": "Pickaxe tools",
+    "extend": { "tools": [ [ [ "pickaxe_plus_one", -1 ] ], [ [ "pickaxe_plus_two", -1 ], [ "eshaper_pickaxe", -1 ] ] ] }
+  },
+  {
+    "id": "mining_standard",
+    "type": "requirement",
+    "//": "Pickaxe tools",
```
Comment (Maleclypse) [inline data/mods/Magiclysm/requirements/crafting.json:16]:
```suggestion
     "//": "mining",
```

File: data/json/requirements/toolsets.json [hunk 11]
```diff
@@ -335,8 +341,7 @@
     "qualities": [ { "id": "HAMMER", "level": 2 }, { "id": "CHISEL", "level": 2 }, { "id": "PRY", "level": 3 } ],
     "tools": [
       [
-        [ "pickaxe", -1 ],
-        [ "bronze_pickaxe", -1 ],
+        [ [ "pickaxe_list", 1, "LIST" ] ],
```
Comment (Maleclypse) [inline data/json/requirements/toolsets.json:344]:
```suggestion
        [ "pickaxe", -1 ],
        [ "bronze_pickaxe", -1 ],
```

File: data/json/requirements/toolsets.json [hunk 12]
```diff
@@ -370,7 +375,7 @@
     "//": "Tools for removing road / sidewalk",
     "qualities": [ { "id": "PRY", "level": 3 } ],
     "tools": [
-      [ [ "pickaxe", -1 ], [ "bronze_pickaxe", -1 ], [ "jackhammer", 140 ], [ "elec_jackhammer", 1848 ] ],
+      [ [ "pickaxe_list", -1, "LIST" ], [ "jackhammer", 140 ], [ "elec_jackhammer", 1848 ] ],
```
Comment (Maleclypse) [inline data/json/requirements/toolsets.json:378]:
```suggestion
      [ [ "pickaxe", -1 ], [ "bronze_pickaxe", -1 ], [ "jackhammer", 140 ], [ "elec_jackhammer", 1848 ] ],
```

File: data/mods/Magiclysm/requirements/crafting.json [hunk 13]
```diff
@@ -3,5 +3,17 @@
...
+    "extend": { "tools": [ [ [ "pickaxe_plus_one", -1 ] ], [ [ "pickaxe_plus_two", -1 ], [ "eshaper_pickaxe", -1 ] ] ] }
+  },
+  {
+    "id": "mining_standard",
+    "type": "requirement",
+    "//": "mining",
+    "extend": { "tools": [ [ [ "pickaxe_plus_one", -1 ] ], [ [ "pickaxe_plus_two", -1 ], [ "eshaper_pickaxe", -1 ] ] ] }
   }
```
Comment (Maleclypse) [inline data/mods/Magiclysm/requirements/crafting.json:18]:
```suggestion
  },
    {
    "id": "concrete_removal_standard",
    "type": "requirement",
    "//": "Tools for removing concrete",
    "extend": { "tools": [ [ [ "pickaxe_plus_one", -1 ] ], [ [ "pickaxe_plus_two", -1 ], [ "eshaper_pickaxe", -1 ] ] ] }
  },
    {
    "id": "road_removal_standard",
    "type": "requirement",
    "//": "Tools for removing road / sidewalk",
    "extend": { "tools": [ [ [ "pickaxe_plus_one", -1 ] ], [ [ "pickaxe_plus_two", -1 ], [ "eshaper_pickaxe", -1 ] ] ] }
  }
```

**Pull request merged**